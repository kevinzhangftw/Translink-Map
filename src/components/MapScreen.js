import React, { Component } from 'react'
import ReactMapGL from 'react-map-gl'
// import mapboxGl from '../css/mapbox-gl.css'

class MapScreen extends Component {

  state = {
    viewport: {
      width: window.innerWidth,
      height: window.innerHeight,
      latitude: 49.2811831,
      longitude: -123.1162378,
      zoom: 12
    }
  }

  render() {
    return (
      <ReactMapGL
        {...this.state.viewport}
        onViewportChange={(viewport) => this.setState({ viewport })}
        mapboxApiAccessToken={process.env.REACT_APP_MAPTOKEN}
      />
    )
  }
}

export default MapScreen
