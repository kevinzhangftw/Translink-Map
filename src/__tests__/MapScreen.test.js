import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import MapScreen from '../components/MapScreen'

it('MapScreen renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<MapScreen />, div)
  ReactDOM.unmountComponentAtNode(div)
})

test('MapScreen renders correct size', () => {
  const component = renderer.create(
    <MapScreen />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
