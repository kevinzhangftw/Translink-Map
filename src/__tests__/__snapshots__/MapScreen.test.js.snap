// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapScreen renders correct size 1`] = `
<div
  style={
    Object {
      "cursor": "-webkit-grab",
      "height": 768,
      "position": "relative",
      "width": 1024,
    }
  }
>
  <div
    style={
      Object {
        "height": 768,
        "position": "relative",
        "width": 1024,
      }
    }
  >
    <div
      style={
        Object {
          "height": 768,
          "visibility": "visible",
          "width": 1024,
        }
      }
    />
    <div
      className="overlays"
      style={
        Object {
          "height": 768,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "top": 0,
          "width": 1024,
        }
      }
    />
  </div>
</div>
`;
